### DISCUSSION TITLE
[Feature] 把关注的人也爬取出来展示

### DISCUSSION THREAD

**Discussion by p0we7** [Chilfish/Weibo-archiver#41] [功能请求]
打算直接销号，如果把关注的人也爬取出来在预览页展示就更好了，微博里也是有一些朋友的。

**Comment by Chilfish** [comment]
对欸，忘了这个，那就只导出关注的人好了

**Reply by p0we7** [reply]
是的，被关注人一般没什么用，大多数是僵尸粉。

**Comment by Chilfish** [comment]
@p0we7 hi，现在可以导出关注列表了🥳🥳可以选择只导出关注列表 [v0.3.11](https://github.com/Chilfish/Weibo-archiver/releases/tag/v0.3.11)

**Reply by p0we7** [reply]
大佬，写得好快啊。我明天试试，多谢大佬

**Comment by p0we7** [comment]
今天终于有空测试了一下，我发现导出的关注列表会忽略我的一些备注，但是PC端好似已经找不懂设置备注的地方，但是依然能看到我的一些备注。

**Reply by Chilfish** [reply]
我还没想过备注的信息欸，我看了一下，这个备注信息能简单拿到，到时有空我就也把这个加进去好了

**Reply by p0we7** [reply]
辛苦了，如果加入之后能通过关注列表搜索就最好。

**Reply by Chilfish** [reply]
现在的 web 端是可以搜索关注列表的呀

![image](https://github.com/Chilfish/Weibo-archiver/assets/60336739/e0afec92-d73a-4118-9c56-bcdb034fc345)

**Reply by p0we7** [reply]
是的，我刚刚有在搜索，我意思是连备注的名字也可以一起搜索。

另外咨询一下，图片的默认CDN选项是什么原理？

**Reply by Chilfish** [reply]
总的来说，这里的CDN就是一种代理转发服务器。微博图片的链接格式是这样的：https://wx3.sinaimg.cn/mw690/00729tAsly1hpo2birw8oj30wu0qqdiv.jpg 。如果直接访问这个链接，会收到没有访问权限的403错误提示。这是因为微博限制了图片只能在其域名下访问，不允许第三方网站访问

因此，可以通过服务器请求该链接，并在HTTP头中伪装成微博的请求，获取到图片文件后，再返回链接

简单的代码实现如下

```ts
async function fetchImage(url: string) {
  const response = await fetch(url, {
    headers: {
      Referer: 'https://weibo.com', // 伪装成微博请求
    },
  });

  return buffer await response.buffer();
}
```

**Reply by p0we7** [reply]
然后 ipfsscan 顺便把图片也缓存下来了，对吗。

**Reply by Chilfish** [reply]
虽然不知道那边是怎么实现的，但也可以这么做。具体的话可以搜一下 ipfs 这个分布式服务器的一些信息，大概就是会存到世界各地的服务器上）

不过为了确保稳定存留，还是下载到本地或是传到自己的图床上较好些，有时候访问频繁了还是什么的，这个cdn会处理不过来。类似的技术可以搜索一下 反代微博图床，可以自行搭建，可以不必全部下载图片

**Reply by p0we7** [reply]
我搜了一下 cdn.ipfsscan.io ，作者有在v2发帖，但是weibo这个接口还是隐藏的，看他github源码 php-api-ipfs 也不存在这个接口

**Reply by Chilfish** [reply]
其实我也不大清楚hhh，我是搜到这个 [2023年微博图床403失效解决办法](https://github.com/Semibold/Weibo-Picture-Store/issues/127#issuecomment-1546823483) 才知道有这个的😹

**Reply by p0we7** [reply]
哈哈，这个其实是作者本人。

**Reply by Chilfish** [reply]
啊注意到了，看来只是没把这个特判写在开源上🤣

**Comment by p0we7** [comment]
自建图床输入链接的输入框好似坏了，我点了也出现不到输入光标。
我 Arc 和 Edge 测试了都是一样的情况，应该出现bug 了。

**Reply by Chilfish** [reply]
对欸不知道什么时候坏的，谢谢提醒😚刚修好它了

**Comment by Chilfish** [comment]
hello，现在可以导出和搜索备注信息了，更新脚本到 0.4.0 版本，然后再重新导出关注列表后到网站上重新导入就行了🥳还是选合并模式

![image](https://github.com/Chilfish/Weibo-archiver/assets/60336739/19e5a348-3e80-4096-8d92-886ef497884b)

**Comment by p0we7** [comment]
@Chilfish 试了一下新版本 0.6.2 ，备注信息又不见了好似。

这个时间确实跨得比较长了，过了一年多了。

**Comment by p0we7** [comment]
你好，我留到0.4.0 做的备注信息版本，到现在0.6.2 版本后找不到了。是打算放弃这个功能了吗？

Chilfish ***@***.***> 于2024年5月15日周三 09:50写道：
